### PR TITLE
DEVX-247 migrate category syncr

### PR DESCRIPTION
**Pull request by lojzatran**
JIRA: https://commercetools.atlassian.net/browse/DEVX-247

I migrated category syncer and its tests. I also commented out other ITs because they're currently not compilable and I don't want to fix them all in this PR.

### CODE REVIEW

**Comment by salander85** [comment]
This was closed accidentially. I will re-open with different base.

File: src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java [hunk 1]
```diff
@@ -216,173 +429,169 @@ private static void deleteProjectData(@Nonnull final SphereClient client) {
...
-            CompletableFuture.allOf(
-                    pageElements.stream()
-                        .map(productType -> deleteProductTypeWithRetry(ctpClient, productType))
-                        .map(CompletionStage::toCompletableFuture)
-                        .toArray(CompletableFuture[]::new))
-                .join();
-
-    CtpQueryUtils.queryAll(ctpClient, ProductTypeQuery.of(), pageConsumer)
+  private static void deleteProductTypesWithRetry(@Nonnull final ProjectApiRoot ctpClient) {
+    // todo: add retry
```
Comment (salander85) [inline src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java:438]: Is there a ticket for this?

File: src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java [hunk 2]
```diff
@@ -216,173 +429,169 @@ private static void deleteProjectData(@Nonnull final SphereClient client) {
...
-                    final Set<UpdateAction<ProductType>> removeActions =
-                        productType.getAttributes().stream()
-                            .map(
-                                attributeDefinition ->
-                                    RemoveAttributeDefinition.of(attributeDefinition.getName()))
-                            .collect(Collectors.toSet());
-                    productTypesToUpdate.put(productType, removeActions);
-                  });
+  private static void deleteProductTypeAttributes(@Nonnull final ProjectApiRoot ctpClient) {
+    QueryUtils.queryAll(
```
Comment (salander85) [inline src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java:460]:
```suggestion
//todo: add retry
    QueryUtils.queryAll(
```

File: src/test/java/com/commercetools/project/sync/category/CategorySyncerTest.java [hunk 3]
```diff
@@ -19,156 +50,215 @@ void setup() {
...
+            .offset(0L)
+            .count(1L)
+            .build();
+    when(response.getBody()).thenReturn(categoryPagedQueryResponse);
+    when(byProjectKeyCategoriesGet.execute())
+        .thenReturn(CompletableFuture.completedFuture(response));
+
+    // test
+    final CategorySyncer categorySyncer =
+        CategorySyncer.of(sourceClient, targetClient, mock(Clock.class));
```
Comment (salander85) [inline src/test/java/com/commercetools/project/sync/category/CategorySyncerTest.java:147]:
```suggestion
        CategorySyncer.of(sourceClient, mock(ProjectApiRoot.class), mock(Clock.class));
```
and remove targetClient variable.

File: src/test/java/com/commercetools/project/sync/category/CategorySyncerTest.java [hunk 4]
```diff
@@ -19,156 +50,215 @@ void setup() {
...
+        categoryPage.stream()
+            .filter(category -> category.getCustom() != null)
+            .map(category -> category.getCustom().getType().getId())
+            .collect(Collectors.toList());
+
+    final String jsonStringCustomTypes =
+        "{\"data\":{\"typeDefinitions\":{\"results\":[{\"id\":\"53c4a8b4-754f-4b95-b6f2-3e1e70e3d0c3\",\"key\":\"cat1\"}]}}}";
+
+    final GraphQLResponse customTypesResult =
+        readObject(jsonStringCustomTypes, GraphQLResponse.class);
```
Comment (salander85) [inline src/test/java/com/commercetools/project/sync/category/CategorySyncerTest.java:90]: You could use `JsonUtils.fromJsonString(jsonString, objectType); `

**Review by salander85 — CHANGES_REQUESTED** [review]

File: src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java (hunk 1 above)
Comment (lojzatran) [reply src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java:438]: https://commercetools.atlassian.net/browse/DEVX-276

File: src/test/java/com/commercetools/project/sync/category/CategorySyncerTest.java (hunk 4 above)
Comment (lojzatran) [reply src/test/java/com/commercetools/project/sync/category/CategorySyncerTest.java:90]: For this part I updated the code in the next PR: https://github.com/commercetools/commercetools-project-sync/pull/499/files#diff-dae00fe4ab31d94681d08aae99801537689b5d162abdaaa50d10a46a25a8d429

File: src/test/java/com/commercetools/project/sync/category/CategorySyncerTest.java (hunk 3 above)
Comment (lojzatran) [reply src/test/java/com/commercetools/project/sync/category/CategorySyncerTest.java]: https://github.com/commercetools/commercetools-project-sync/pull/498/commits/db7224267c69019001a9ea6c16f38fa021c87a5a

File: src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java (hunk 2 above)
Comment (lojzatran) [reply src/integration-test/java/com/commercetools/project/sync/util/IntegrationTestUtils.java:461]: https://github.com/commercetools/commercetools-project-sync/pull/498/commits/3ffc9cf1ead2dcb2ffceae1dd9f80327b93a1a3e

**Review by salander85 — APPROVED** [review]
Good work!

**Pull request closed without merging**